### PR TITLE
Add component for form fields

### PR DESCRIPTION
**Pull request by juhoeerola**
The component wraps:
- optional label
- input element
- error message for the input

### CODE REVIEW

**Comment by IidaKainu** [comment]
The error messages look great and function well
I noticed the errors activate after the form is first submitted. A lot of forms I've encountered show the errors as user is typing. Could be beneficial to ask Teemu how they are intended to work here (probably hard to demonstrate with figma)

**Comment by juhoeerola** [comment]
> I noticed the errors activate after the form is first submitted

Once the specific field has been invalid once (on submit), it shows the validation status while typing (except validation from server, or submit handler of course).

I went with this approach, as it would be annoying if all the fields would nag about validation errors from the start.

Maybe validating a field when focus leaves it, would be an option.

**Comment by IidaKainu** [comment]
Yeah I can see how it could be annoying. Validating the field on focus leave seems like an good option. At least it could decrease the amount of errors user encounters at the same time.

File: frontend/src/components/forms/FormField.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+function field<T extends FieldType>(
+  Tag: React.ComponentType<React.HTMLProps<T>>,
+): React.FC<FieldProps<T>> {
+  return ({ error, label, id, ...props }) => {
+    const [localError, setLocalError] = useState('');
+    const err = error ?? {
+      message: localError,
+      setMessage: setLocalError,
+    };
```
Comment (IidaKainu) [inline frontend/src/components/forms/FormField.tsx:45]: Just out of curiosity: why both localError and err are used as localError is only used in err?

**Review by IidaKainu — CHANGES_REQUESTED** [review]
I won't pretend to completely understand how the FormField works, but in testing it functions just as planned!

Labels could be added to milestone, task creation and task edition  -fields for consistency
Other than that, looks good 👍

File: frontend/src/components/modals/EditVersionModal.tsx [hunk 2]
```diff
@@ -64,12 +66,13 @@ export const EditVersionModal: React.FC<EditVersionModalProps> = ({
             <label htmlFor="name">
               <Trans i18nKey="Milestone name" />
             </label>
```
[the remark this answers is not in the document]
Comment (juhoeerola) [reply frontend/src/components/modals/EditVersionModal.tsx]: good catch

File: frontend/src/components/forms/FormField.tsx (hunk 1 above)
Comment (juhoeerola) [reply frontend/src/components/forms/FormField.tsx]: I think I had some linter error/warning about conditionally calling `useState`, but didn't get it anymore

Cleaned this up a little now

**Review by IidaKainu — APPROVED** [review]

**Pull request merged**